src/main/java/teammates/ui/webapi/PutDataBundleDocumentsAction.java

```diff
@@ -27,8 +28,10 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
     @Override
     public JsonResult execute() {
         DataBundle dataBundle = JsonUtils.fromJson(getRequestBody(), DataBundle.class);
+        SqlDataBundle sqlDataBundle = JsonUtils.fromJson(getRequestBody(), SqlDataBundle.class);
         try {
             logic.putDocuments(dataBundle);
+            sqlLogic.putDocuments(sqlDataBundle);
         } catch (SearchServiceException e) {
```

Comment:
Not entirely sure about this... if I'm not wrong, the flow would be:
- Request body is of type SqlDataBundle
- Error thrown for `JsonUtils.fromJson(getRequestBody(), DataBundle.class)` since the 2 classes cannot be cast to each other (?)
- Uncaught error -> sqlDataBundle is never called with method `sqlLogic.putDocuments`

Similar story if request body is of type `DataBundle`

lmk if your understanding is different

Reply:
yea you are right an error is thrown as the 2 classes cannot be cast to each other

I have made the following changes:

- add a `databundletype` param to the action
- update `putDocuments` method in `AbstractBackDoor`to have the new param in the request
- create `putSqlDocuments` method in `AbstractBackDoor`

I have the param and types hard coded instead of being added into `Const` as this action is only used by `AbstractBackDoor`(?) so that clean up is easier after migration or should I add them into `Const`?